### PR TITLE
HeadFetcher data race fix

### PR DESCRIPTION
**Pull request by skillful-alex**
I started the beacon-chain with the data race detection, got an error, and fixed it.
Error:
```
======================================
WARNING: DATA RACE
Read at 0x00c000846510 by goroutine 41:
  github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).HeadState()
      beacon-chain/blockchain/chain_info.go:76 +0x45
  github.com/prysmaticlabs/prysm/beacon-chain/rpc.(*ValidatorServer).MultipleValidatorStatus()
      beacon-chain/rpc/validator_server.go:248 +0x7b
  github.com/prysmaticlabs/prysm/beacon-chain/rpc.(*ValidatorServer).WaitForActivation()
      beacon-chain/rpc/validator_server.go:45 +0xd6
...
Previous write at 0x00c000846510 by goroutine 131:
  github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).saveGenesisData()
      beacon-chain/blockchain/service.go:267 +0x9aa
  github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).initializeBeaconChain()
      beacon-chain/blockchain/service.go:163 +0x250
  github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).processChainStartTime()
      beacon-chain/blockchain/service.go:137 +0x13b
  github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).Start.func1()
      beacon-chain/blockchain/service.go:127 +0xc9
...
===============================================
WARNING: DATA RACE
Read at 0x00c000846508 by goroutine 138:
  github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).HeadBlock()
      beacon-chain/blockchain/chain_info.go:71 +0x45
  github.com/prysmaticlabs/prysm/beacon-chain/rpc.(*BeaconServer).CanonicalHead()
      beacon-chain/rpc/beacon_server.go:73 +0x55
...
Previous write at 0x00c000846508 by goroutine 131:
  github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).saveGenesisData()
      beacon-chain/blockchain/service.go:266 +0x96e
  github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).initializeBeaconChain()
      beacon-chain/blockchain/service.go:163 +0x250
  github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).processChainStartTime()
      beacon-chain/blockchain/service.go:137 +0x13b
  github.com/prysmaticlabs/prysm/beacon-chain/blockchain.(*Service).Start.func1()
      beacon-chain/blockchain/service.go:127 +0xc9
...
```

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Looks like there are still data races in the CI

**Comment by nisdas** [comment]
seems like tests are still failing @skillful-alex , the race detector has detected races in other tests. Maybe you can apply race only for this test ? For reference you can take a look here on how
to apply race detection to a subset of tests.
https://github.com/prysmaticlabs/prysm/blob/b04b542e645752f7b8c6209f2f5d984ef62dbbc7/shared/deprecated-p2p/BUILD.bazel#L115

**Comment by skillful-alex** [comment]
The error occurs if the build options --collect_code_coverage and race="on" are active at the same time. 
In CI step: bazel --bazelrc=.buildkite-bazelrc coverage //... --features=norace --test_tag_filters="-spectest"

**Comment by skillful-alex** [comment]
Can we add to buildkite step 
`bazel --bazelrc=.buildkite-bazelrc coverage //... --features=norace --test_tag_filters="-spectest"
`test_tag_filters "-race_on"?

To make the step:
`bazel --bazelrc=.buildkite-bazelrc coverage //... --features=norace --test_tag_filters="-spectest,-race_on"
`

**Comment by skillful-alex** [comment]
Thanks for the changes. Now CI works without errors.
Data race tests are not used when calculating coverage.

File: beacon-chain/blockchain/receive_attestation.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ func (s *Service) ReceiveAttestationNoPubsub(ctx context.Context, att *ethpb.Att
 
 	// Skip checking for competing attestation's target roots at epoch boundary.
 	if !helpers.IsEpochStart(attSlot) {
-		targetRoot, err := helpers.BlockRoot(s.headState, att.Data.Target.Epoch)
+		targetRoot, err := helpers.BlockRoot(s.HeadState(), att.Data.Target.Epoch)
```
Comment (nisdas) [inline beacon-chain/blockchain/receive_attestation.go]: lets not clone the state here, we are only reading from the state

Comment (skillful-alex) [reply beacon-chain/blockchain/receive_attestation.go]: fixed

**Pull request merged**